Example/ViewController.swift

```diff
@@ -44,13 +50,229 @@ class ViewController: UIViewController {
+
+        shapeLayer.textColor = NSExpression(forConditional: NSPredicate(format: "selected == true"),
+                                            trueExpression: NSExpression(forConstantValue: UIColor.white),
+                     falseExpression: NSExpression(forConstantValue: UIColor.black))
+
+        shapeLayer.textFontNames = NSExpression(forConstantValue: ["DIN Pro Medium"])
+        shapeLayer.textAllowsOverlap = NSExpression(forConstantValue: true)
+        shapeLayer.textJustification = NSExpression(forConstantValue: "left")
+        shapeLayer.symbolZOrder = NSExpression(forConstantValue: NSValue(mglSymbolZOrder: MGLSymbolZOrder.auto))
+        shapeLayer.symbolSortKey = NSExpression(forConditional: NSPredicate(format: "selected == true"),
+                                                trueExpression: NSExpression(forConstantValue: 1),
+                                                   falseExpression: NSExpression(format: "sortOrder"))
+        shapeLayer.iconAnchor = NSExpression(forConditional: NSPredicate(format: "tailPosition == 0"),
+                                             trueExpression: NSExpression(forConstantValue: "bottom-left"),
+                                                falseExpression: NSExpression(forConstantValue: "bottom-right"))
+        shapeLayer.textAnchor = shapeLayer.iconAnchor
+        shapeLayer.iconTextFit = NSExpression(forConstantValue: "both")
+
+        shapeLayer.iconImageName = NSExpression(forKeyPath: "imageName")
+        shapeLayer.iconOffset = NSExpression(forConditional: NSPredicate(format: "tailPosition == 0"),
+                                             trueExpression: NSExpression(forConstantValue: CGVector(dx: 0.5, dy: -1.0)),
+                      falseExpression: NSExpression(forConstantValue: CGVector(dx: -0.5, dy: -1.0)))
+        shapeLayer.textOffset = shapeLayer.iconOffset
+        shapeLayer.iconAllowsOverlap = NSExpression(forConstantValue: true)
+
+        style.addLayer(shapeLayer)
+    }
+
+    private func removeRouteAnnotationsLayerFromStyle(_ style: MGLStyle) {
+        if let annotationsLayer = style.layer(withIdentifier: annotationLayerIdentifier + "-shape") {
+            style.removeLayer(annotationsLayer)
+        }
+
+        if let annotationsSource = style.source(withIdentifier: annotationLayerIdentifier + "-source") {
+            style.removeSource(annotationsSource)
         }
     }
-    
+
+    // This function generates the text for the label to be shown on screen. It will include estimated duration and info on Tolls, if applicable
+    private func annotationLabelForRoute(_ route: Route, tolls: Bool) -> String {
+        var eta = DateComponentsFormatter.shortDateComponentsFormatter.string(from: route.expectedTravelTime) ?? ""
+
+        let hasTolls = (route.tollIntersections?.count ?? 0) > 0
+        if hasTolls {
+            eta += "\n" + NSLocalizedString("ROUTE_HAS_TOLLS", value: "Tolls", comment: "This route does have tolls")
```

Comment:
Remember to run scripts/extract_localizable.sh to update the Base localization.

Reply:
Done.